ote/src/clj/ote/services/transport.clj

```diff
@@ -247,17 +248,41 @@
                        (filter (comp (complement from-ui) ::t-service/id) from-db))]
     to-delete))
 
+(defn- delete-external-companies
+  "User might remove url from service, so we delete all service-companies from db"
+  [db transport-service]
+    (specql/delete! db ::t-service/service-company {::t-service/transport-service-id (::t-service/id transport-service)}))
+
+(defn- save-external-companies
+  "Service can contain an url that contains company nmes and business-id. Sevice can also contain an imported csv file
+  with company names and business-ids."
+  [db transport-service]
+  (let [current-data (first (fetch db ::t-service/service-company (specql/columns ::t-service/service-company)
+                            {::t-service/transport-service-id (::t-service/id transport-service)}))
+        companies (into [] (:companies (external/check-csv db {:url (get transport-service ::t-service/companies-csv-url)
+                                                                            :ts-id (get transport-service ::t-service/id)})))
+        new-data (if (empty? current-data)
+                      {::t-service/companies            companies
+                       ::t-service/transport-service-id (get transport-service ::t-service/id)
+                       ::t-service/source               "URL"}
+                      (assoc current-data ::t-service/companies companies))]
+
+    (external/save-companies db new-data)
+  ))
+
 (defn- save-transport-service
   "UPSERT! given data to database. And convert possible float point values to bigdecimal"
   [nap-config db user {places ::t-service/operation-area
                        external-interfaces ::t-service/external-interfaces
+                       service-company ::t-service/service-company
                        :as data}]
   ;(println "DATA: " (pr-str data))
   (let [service-info (-> data
                          (modification/with-modification-fields ::t-service/id user)
                          (dissoc ::t-service/operation-area)
                          floats-to-bigdec
-                         (dissoc ::t-service/external-interfaces))
+                         (dissoc ::t-service/external-interfaces)
```

Comment:
Combine these two dissocs with the above, 1 `dissoc` call can take multiple keys.